feat(ia): audience campaigns

miguelpeixe opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

1205919985867982-as-1206833421212801/f

Moves the Campaigns wizard to Audience -> Campaigns.

<img width="1364" alt="image" src="https://github.com/user-attachments/assets/eef63713-1920-45a5-960c-34d27ffb5d87">

<img width="1067" alt="image" src="https://github.com/user-attachments/assets/af4edd95-9234-40a2-aa41-30b18354d945">

<img width="1365" alt="image" src="https://github.com/user-attachments/assets/8d7d9c41-4b83-47e8-a455-767c5472b80d">

<img width="1367" alt="image" src="https://github.com/user-attachments/assets/0f2604ea-172d-4bb5-84e6-883e8917afe7">

<img width="1365" alt="image" src="https://github.com/user-attachments/assets/a5a51153-e3c9-4f35-a8fd-aa4d822aebed">

### How to test the changes in this Pull Request:

#### Campaigns tab

1. Confirm you have RAS setup (via Audience -> Configuration)
2. Checkout this branch, https://github.com/Automattic/newspack-popups/pull/1365, and visit Audience -> Campaigns
3. Confirm the Campaigns tab renders as expected (see image above)
4. Click "Preview Segment" on any of the segments and confirm the preview renders with the expected prompts
5. Click on a prompt settings button and confirm the modal renders
6. Tweak the settings, save, and confirm the changes persist
7. Click a prompt's "more" button, then "Edit"
8. Confirm the editor renders with the correct sidebar menu active
9. Back to the wizard, click on a prompt's "more" button, then "Preview", and confirm the prompt preview renders as expected
10. Do the same with "Duplicate", "Deactivate", and "Delete"
11. Click "Add New Campaign", add the campaign, and confirm the campaign is created
12. Add a new prompt to a segment, confirm the correct menu is active, save the prompt, and go back to the wizard
13. Confirm the created prompt renders for the campaign
14. Switch the view from "Active Prompts" to all the other options and confirm the filters are applied correctly

#### Segments tab

1. Switch to the Segments tab and confirm it renders as expected (see image above)
2. Toggle segments and confirm the changes persist
3. Use the chevrons to move their priority and confirm it also persists
4. Edit a segment and confirm the changes persist
5. Create a new segment and confirm it's created
6. Delete the created segment and confirm it's deleted

#### Analytics tab

1. Confirm the tab renders as expected and the links go to the correct destination

#### Settings tab

1. Confirm the tab renders as expected (image above)
2. Change all the options to something else and save
3. Confirm the changes persist

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

miguelpeixe committed:
chore: move wizard js

miguelpeixe committed:
fix: tweak margins and paddings

miguelpeixe committed:
chore: move api endpoint and delete popups wizard

miguelpeixe committed:
fix: active menu

miguelpeixe committed:
fix: audience campaigns type and export wizard

miguelpeixe committed:
fix: segments tab active when editing segment

miguelpeixe committed:
chore: directory structure

miguelpeixe committed:
chore: rename component

dkoo reviewed (changes requested):
@miguelpeixe Functionally, working really well! I noticed a couple of tiny style bugs, one of which isn't even related to these changes (but since we're here, may as well fix it):

1. If a segment contains no prompts in the current view, the styles look a little off. Looks like we're missing some padding, compared to `trunk`:

| `trunk` | this branch |
| --- | --- |
| <img width="1084" alt="Screenshot 2024-11-19 at 4 51 12 PM" src="https://github.com/user-attachments/assets/fbde0e1e-f428-4149-bf4c-a3aada3f19e3"> | <img width="1124" alt="Screenshot 2024-11-19 at 4 37 30 PM" src="https://github.com/user-attachments/assets/33434be6-f457-4216-a48a-8c0914dbfb80"> |

2. This is on `trunk` too, but now's a good time to fix it. The dropdown for the filters seems to be missing some CSS. Note the unnecessary padding in the highlighted/unselected items, and the extra black checkmark in the selected item.

<img width="180" alt="Screenshot 2024-11-19 at 4 43 39 PM" src="https://github.com/user-attachments/assets/d497a7d8-bbbc-45cf-9afb-552ea12f261f">



Otherwise looking really clean 👍

miguelpeixe committed:
fix: campaign filter styles

miguelpeixe committed:
fix: empty segment text styles

miguelpeixe commented:
Thank you for the review, @dkoo!

> If a segment contains no prompts in the current view, the styles look a little off. Looks like we're missing some padding, compared to trunk:

Fixed in a236248778b867e290a35bee660795820b7daaee:

<img width="866" alt="image" src="https://github.com/user-attachments/assets/777b4bd8-4a37-4c27-980d-d433de7cc53c">

> This is on trunk too, but now's a good time to fix it. The dropdown for the filters seems to be missing some CSS. Note the unnecessary padding in the highlighted/unselected items, and the extra black checkmark in the selected item.

Good call! Fixed in 50342d9d832989e199735d6641d80fbc15fd6194

<img width="243" alt="image" src="https://github.com/user-attachments/assets/42100a73-a4fc-4453-bf04-6e9d1b2ee70f">

I've decided to use the component's native icon with adjusted color instead of our custom pseudo-element. I also added a min-width to the list so it looks better when the selected filter is a shorter word.

dkoo commented on includes/wizards/audience/class-audience-campaigns.php:
```diff
@@ -63,4 +114,929 @@ public function add_page() {
+	}
+
+	/**
+	 * Update settings for a Pop-up.
```
```suggestion
	 * Update settings for a prompt.
```

ABSOLUTELY UNNECESSARY NIT: is now a good time to do the ol' Find/Replace to rename `pop-up` to `prompt`? 😄

miguelpeixe replied:
Can't think of a better time! Done in 3a06b084b3b99934c97231d85e7288e00c434711.

I've left the CM methods unchanged since they are wrappers for the plugin's methods. The plugin still uses `popup`.

I also removed the `api_set_popup_terms()` which was unused (also unused in `trunk`).

dkoo reviewed (approved):
Looking really nice—thanks for the updates, @miguelpeixe!

miguelpeixe committed:
chore: replace `popup` to `prompt`
